### PR TITLE
feat(help): ban banned people from creating threads

### LINKED ISSUE #94: [help] Reject threads from banned users
The help ban role (`883649900913512508`) should not be able to create threads.

This will allow the permission overrides for that role to change in the following way:
```diff
+  view channel
-  send messages in threads
```
That will allow banned users to still view their past threads as reference.

### PR DESCRIPTION
**Pull request by MaskDuck**
close #94

### CODE REVIEW

File: cogs/help.py [hunk 1]
```diff
@@ -228,6 +229,13 @@ def __init__(self, bot: commands.Bot):
             HelpButton("Nextcord", style=ButtonStyle.blurple, custom_id="nextcord")
         )
         self.add_item(HelpButton("Python", style=ButtonStyle.green, custom_id="python"))
+    
+    async def interaction_check(self, interaction: Interaction):
+        if interaction.user.get_role(883649900913512508) is not None:
+            await interaction.send(NO_HELP_MESSAGE, ephemeral=True)
+            return False
+
+        return await super().interaction_check(interaction)
```
Comment (onerandomusername) [inline cogs/help.py:238]: should just return true

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: cogs/help.py [hunk 2]
```diff
@@ -34,6 +34,7 @@
 CUSTOM_ID_PREFIX: str = "help:"
 NAME_TOPIC_REGEX: str = r"^(?P<topic>.*?) \((?P<author>[^)]*[^(]*)\)$"
 WAIT_FOR_TIMEOUT: int = 1800  # 30 minutes
+NO_HELP_MESSAGE: str = "You are banned from creating help threads."
```
Comment (koala9712) [inline cogs/help.py:37]: I feel this can be elaborated a bit

File: cogs/help.py [hunk 3]
```diff
@@ -228,6 +229,13 @@ def __init__(self, bot: commands.Bot):
             HelpButton("Nextcord", style=ButtonStyle.blurple, custom_id="nextcord")
         )
         self.add_item(HelpButton("Python", style=ButtonStyle.green, custom_id="python"))
+    
+    async def interaction_check(self, interaction: Interaction):
+        if interaction.user.get_role(883649900913512508) is not None:
```
Comment (ooliver1) [inline cogs/help.py:234]: Unsure if this could be done like the others with a constant at the top of this file

File: cogs/help.py (hunk 2 above)
Comment (koala9712) [reply cogs/help.py:38]: Why was this marked as resolved without any action?
Please give a response before marking reviews as resolved.

Comment (MaskDuck) [reply cogs/help.py:38]: wait somehow i misclicked :(

Comment (Skelmis) [reply cogs/help.py:38]: I agree, possibly some form of action to take if they feel its unfair. Like dm'ing modmail

File: cogs/help.py [hunk 4]
```diff
@@ -30,10 +30,12 @@
 HELP_LOGS_CHANNEL_ID: int = int(env["HELP_LOG_CHANNEL_ID"])
 HELPER_ROLE_ID: int = int(env["HELP_NOTIFICATION_ROLE_ID"])
 HELP_MOD_ID: int = int(env["HELP_MOD_ROLE_ID"])
+NO_HELP_ID: int = int(env["NO_HELP_ROLE_ID"])
```
Comment (realShamlol) [inline cogs/help.py:33]: Could this be renamed to `HELP_BANNED_ID` just to be more consistent with the Nextcord server role names.

Comment (koala9712) [reply cogs/help.py:33]: `HELP_BAN_ID` hehe
SCREAMING SNAKE CASE ftw,

Comment (toolifelesstocode) [reply cogs/help.py:33]: > `HELP_BAN_ID` hehe SCREAMING SNAKE CASE ftw,

🙏🏼

File: cogs/help.py [hunk 5]
```diff
@@ -30,7 +30,7 @@
 HELP_LOGS_CHANNEL_ID: int = int(env["HELP_LOG_CHANNEL_ID"])
 HELPER_ROLE_ID: int = int(env["HELP_NOTIFICATION_ROLE_ID"])
 HELP_MOD_ID: int = int(env["HELP_MOD_ROLE_ID"])
-NO_HELP_ID: int = int(env["NO_HELP_ROLE_ID"])
+HELP_BAN_ID: int = int(env["HELP_BANNED_ROLE_ID"])
```
Comment (toolifelesstocode) [inline cogs/help.py:33]: This is `HELP_BAN_ID`

File: cogs/help.py [hunk 6]
```diff
@@ -232,7 +232,7 @@ def __init__(self, bot: commands.Bot):
         self.add_item(HelpButton("Python", style=ButtonStyle.green, custom_id="python"))
     
     async def interaction_check(self, interaction: Interaction):
-        if interaction.user.get_role(NO_HELP_ID) is not None:
+        if interaction.user.get_role(HELP_BANNED_ID) is not None:
```
Comment (toolifelesstocode) [inline cogs/help.py:235]: ....And this is `HELP_BANNED_ID`, which will lead to python saying that this is undefined.

**Review by toolifelesstocode — CHANGES_REQUESTED** [review]

**Review by toolifelesstocode — APPROVED** [review]

**Review by koala9712 — APPROVED** [review]

**Review by Soheab — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**